src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -1,22 +1,62 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
 import {createDrawerNavigator} from '@react-navigation/drawer';
+import {withOnyx} from 'react-native-onyx';
 
 import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
 import styles, {
     getNavigationDrawerType,
     getNavigationDrawerStyle,
 } from '../../../styles/styles';
+import ONYXKEYS from '../../../ONYXKEYS';
+import compose from '../../compose';
+import SCREENS from '../../../SCREENS';
 
 // Screens
 import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import ReportScreen from '../../../pages/home/ReportScreen';
 
+const LoadingScreen = React.memo(() => <FullScreenLoadingIndicator visible />, () => true);
+
 const propTypes = {
+    // Initial report to be used if nothing else is specified by routing
+    initialReportID: PropTypes.string,
+
+    // Route data passed by react navigation
+    route: PropTypes.shape({
+        state: PropTypes.shape({
+            // A list of mounted screen names
+            routeNames: PropTypes.arrayOf(PropTypes.string),
+        }),
+    }).isRequired,
+
     ...windowDimensionsPropTypes,
 };
 
+const defaultProps = {
+    initialReportID: null,
+};
+
 const Drawer = createDrawerNavigator();
 
+/**
+ * Derives whether it's time to mount the ReportScreen from current component props
+ *
+ * @param {String} initialReportID
+ * @param {Object} route
+ * @returns {boolean}
+ */
+const shouldMountReportScreen = ({initialReportID, route}) => {
+    if (_.isString(initialReportID)) {
+        return true;
+    }
+
+    // After the ReportScreen is mounted it should stay mounted no matter initial report ID
+    return Boolean(route.state) && route.state.routeNames.includes(SCREENS.REPORT);
+};
+
```

Comment:
This could be put in the component body like

```jsx
const MainDrawerNavigator = (props) => {
    const hasInitialParams = _.isString(props.initialReportID);
    const const isReportMounted = Boolean(route.state) && route.state.routeNames.includes(SCREENS.REPORT);

   return (
        <Drawer.Navigator>
           {
               hasInitialParams || isReportMounted 
                  ? <Drawer.Screen name={SCREENS.REPORT} />
                  : <Drawer.Screen name={SCREENS.LOADING} />
            }
    );
```

---

Underscore `get` function would have been useful here

```js
return _.get(route, ['state', 'routeNames'], []).includes(SCREENS.REPORT);
```

But it's not available in the current version

Reply:
I've just noticed there's a warning regarding accessing state
> Accessing the 'state' property of the 'route' object is not supported. If you want to get the focused route name, use the 'getFocusedRouteNameFromRoute' helper instead: https://reactnavigation.org/docs/screen-options-resolution/#setting-parent-screen-options-based-on-child-navigators-state

I'll see if it's possible to change this so this is not using `route.state`